docs/reference/how-to/size-your-shards.asciidoc

```diff
@@ -0,0 +1,308 @@
+particular indices only move to specified nodes.
+
+[discrete]
+[[shard-size-guidelines]]
+=== Guidelines
+
+Where applicable, use the following guidelines as starting points for your
+sharding strategy.
+
+[discrete]
+[[delete-indices-not-documents]]
+==== Delete indices, not documents
+
+Deleting individual documents often increases shard overhead. Deleted documents
+aren't immediately removed from the file system. Instead, {es} marks the
+document as deleted on each related shard. The marked document will continue to
+use resources until it's removed during a periodic <<index-modules-merge,segment
+merge>>.
+
+When possible, delete entire indices instead. {es} can immediately remove
+deleted indices directly from the file system and free up resources.
+
+[discrete]
+[[use-ds-ilm-for-time-series]]
+==== For time series data, use data streams and {ilm-init}
+
+Data streams let you store time series data across multiple, time-based backing
+indices. You can use <<index-lifecycle-management,{ilm} ({ilm-init})>>
+lifecycle policies to automatically manage these backing indices.
+
+One advantage of this setup is
+<<getting-started-index-lifecycle-management,automatic rollover>>, which creates
+a new write index when the current one meets a defined `max_age`, `max_docs`, or
+`max_size` threshold. You can use these thresholds to create indices based on
+your retention intervals. When an index is no longer needed, you can use
+{ilm-init} to automatically delete it and free up resources.
+
+{ilm-init} also makes it easy to change your sharding strategy over time:
+
+* *Want to decrease the shard count for new indices?* +
+Change the <<index-number-of-shards,`index.number_of_shards`>> setting in the
+data stream's <<data-streams-change-mappings-and-settings,matching index
+template>>.
+
+* *Want larger shards?* +
+Increase the rollover threshold in your {ilm-init} policy.
+
+* *Need indices that span shorter intervals?* +
+Offset the increased shard count by deleting older indices sooner.
+
+Every new backing index is an opportunity to further tune your strategy.
+
+[discrete]
+[[shard-size-recommendation]]
+==== Aim for shard sizes between 10GB and 50GB
+
+Shards larger than 50GB may make a cluster less likely to recover from failure.
+When a node fails, {es} rebalances the cluster's shards across its remaining
+nodes. Shards larger than 50GB can be harder to move across a network and may
+tax node resources.
+
+[discrete]
+[[shard-count-recommendation]]
+==== Aim for 20 shards or fewer per GB of heap space
+
+The number of shards a node can hold is proportional to the node's
+<<heap-size,heap space>>. For example, a node with a 30GB heap should have at
+most 600 shards. The further below this limit you can keep your nodes, the
+better. If you find your nodes exceeding more than 20 shards per GB, consider
+adding another node. You can use the cat shards API to check the number of
+shards per node.
```

Comment:
Note for reviewers:

Not sure if this guideline still hold now that heap usage is reduced:
https://www.elastic.co/guide/en/elasticsearch/reference/7.7/release-highlights.html#_significant_reduction_of_heap_usage_of_segments

Any update guidance is appreciated!